app/models/file_search.rb

```diff
@@ -1,10 +1,13 @@
 require "solrizer"
 
 class FileSearch
-  def initialize(params, catalog_query:, file_query: GenericFile)
+  attr_accessor :current_page, :total_pages, :next_page, :total_items
+
+  def initialize(params, options = {} )
     @params = params
-    @catalog_query = catalog_query
-    @file_query = file_query
+    @resource_type = options[:resource_type]
+    @catalog_query = options[:catalog_query]
+    @file_query = options[:file_query] || GenericFile
```

Comment:
Is this better than using keyword arguments?  I'm not sure; I tend to prefer kw args instead, but that might be my inner Smalltalker speaking.